hybrid/include/utils.h

```diff
@@ -0,0 +1,25 @@
+/* Inline Assembly utility functions to read/write to/from
+ * the Default Data Capability Register (DDC)
+ */
+
+void write_ddc(void *__capability cap);
+void *__capability read_ddc();
+
+/* Write a capability to the DDC
+ */
+inline void write_ddc(void *__capability cap)
+{
+	asm("MSR DDC, %[cap]\n\t"
+		:
+		: [cap] "C"(cap) /* input regs */
+		:);
```

Comment:
I think you want a "memory" clobber here, since you're writing a system register that the compiler doesn't know about.

As discussed, this gets complicated in general (e.g. Laurie's benchmarking example) but it works in my local test framework.

Reply:
Alternatively (and I strongly recommend this here), make this whole file `utils.s` assembly file and just rule out compiler interference that way.

Reply:
I tried "memory" yesterday but it didn't work. Your `utils.s` solution maybe it is a better way to address this issue. Assuming, there is no other interference or problem `CHERI/builtin`-related.

Reply:
I was wrong, this now works in 1ae5ff2.